interfaces/builtin/docker_support.go

```diff
@@ -161,12 +161,172 @@ pivot_root,
+change_profile unsafe /snap/snap[^d]** -> docker-default,
+change_profile unsafe /snap/snapd[^/]** -> docker-default,
+change_profile unsafe /snap/snapd/*/[^u]** -> docker-default,
+change_profile unsafe /snap/snapd/*/u[^s]** -> docker-default,
+change_profile unsafe /snap/snapd/*/us[^r]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr[^/]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/[^l]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/l[^i]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/li[^b]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib[^/]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/[^s]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/s[^n]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/sn[^a]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/sna[^p]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/snap[^d]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/snapd[^/]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/snapd/[^s]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/snapd/s[^n]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/snapd/sn[^a]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/snapd/sna[^p]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap[^-]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-[^c]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-c[^o]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-co[^n]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-con[^f]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-conf[^i]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-confi[^n]** -> docker-default,
+change_profile unsafe /snap/snapd/*/usr/lib/snapd/snap-confin[^e]** -> docker-default,
+
+
+# signal/tracing rules too
 signal (send) peer=docker-default,
 ptrace (read, trace) peer=docker-default,
 
+
 # defaults for containerd
-change_profile unsafe /** -> cri-containerd.apparmor.d,
+# TODO: When we drop support for executing other snaps from devmode snaps (or 
+# when the AppArmor parser bugs are fixed) this can go back to the much simpler
+# rule:	
+# change_profile unsafe /** -> cri-containerd.apparmor.d,
+# see above comment, we need this because we can't have nice things
+change_profile unsafe /[^s]** -> cri-containerd.apparmor.d,
+change_profile unsafe /s[^n]** -> cri-containerd.apparmor.d,
+change_profile unsafe /sn[^a]** -> cri-containerd.apparmor.d,
+change_profile unsafe /sna[^p]** -> cri-containerd.apparmor.d,
+change_profile unsafe /snap[^/]** -> cri-containerd.apparmor.d,
+change_profile unsafe /snap/{s[^n],c[^o]}** -> cri-containerd.apparmor.d,
```

Comment:
As above - do we need a line above this of the form:

```
 change_profile unsafe /snap/{[^s],[^c]}** -> cri-containerd.apparmor.d,
```

Reply:
done